packages/beeq/src/components/progress/bq-progress.tsx

```diff
@@ -0,0 +1,158 @@
+import { validatePropValue } from '../../shared/utils';
+
+@Component({
+  tag: 'bq-progress',
+  styleUrl: './scss/bq-progress.scss',
+  shadow: true,
+})
+export class BqProgress {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqProgressElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** If `true` the indeterminate state of progress bar is enabled */
+  @Prop({ reflect: true }) indeterminate: boolean = false;
+
+  /** A number representing the current value of the progress bar */
+  @Prop({ reflect: true }) value = 0;
+
+  /** Progress bar thickness */
+  @Prop({ reflect: true }) thickness: TProgressThickness = 'medium';
+
+  /** Progress type */
+  @Prop({ reflect: true }) type: TProgressType = 'default';
+
+  /** If `rounded`, the progress bar will be displayed without border radius */
+  @Prop({ reflect: true }) borderShape: TProgressBorderShape = 'rounded';
+
+  /** If `true`, the progress bar will be displayed with percentage text */
+  @Prop({ reflect: true }) label: boolean = false;
+
+  /** If `true`, the progress bar will be displayed with percentage tooltip */
+  @Prop({ reflect: true }) enableTooltip: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('borderShape')
+  @Watch('thickness')
+  @Watch('type')
+  handleTypePropChange() {
+    validatePropValue(PROGRESS_BORDER_SHAPE, 'rounded', this.el, 'borderShape');
+    validatePropValue(PROGRESS_THICKNESS, 'medium', this.el, 'thickness');
+    validatePropValue(PROGRESS_TYPE, 'default', this.el, 'type');
+  }
+
+  @Watch('value')
+  handleValuePropChange(newValue: number) {
+    this.validateValue(newValue);
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.handleTypePropChange();
+    this.handleValuePropChange(this.value);
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private validateValue(newValue: number) {
+    if (this.value) {
+      const clampedValue = Math.max(0, Math.min(100, newValue));
+      if (newValue !== clampedValue) {
+        this.value = clampedValue;
+      }
+    }
+    return this.value;
+  }
```

Comment:
The method can be improved by removing the unnecessary `if` condition. The `if` condition checks if `this.value` is truthy, but it doesn't make sense in this context because we are validating the `newValue`, not `this.value`.

```suggestion
  private validateValue(newValue: number) {
    const clampedValue = Math.max(0, Math.min(100, newValue));
    if (newValue !== clampedValue) {
      this.value = clampedValue;
    }
    return this.value;
  }
```